packages/assets-controllers/src/TokenRatesController.test.ts

```diff
@@ -120,19 +108,27 @@ describe('TokenRatesController', () => {
     const fetchSpy = jest.spyOn(globalThis, 'fetch').mockImplementation(() => {
       throw new Error('Network error');
     });
+    let tokenStateChangeListener: (state: any) => void;
+    const onTokensStateChange = sinon.stub().callsFake((listener) => {
+      tokenStateChangeListener = listener;
+    });
     const interval = 100;
     const controller = new TokenRatesController(
       {
         chainId: toHex(1),
         ticker: NetworksTicker.mainnet,
-        onTokensStateChange: jest.fn(),
+        onTokensStateChange,
         onNetworkStateChange: jest.fn(),
       },
       {
         interval,
-        tokens: [{ address: 'bar', decimals: 0, symbol: '', aggregators: [] }],
```

Comment:
I made this change because I started migrating the tests with `token` setters to set the initial configuration instead, but noticed that it wasn't working. It turns out that the initial configuration gets overwritten by the default configuration, so this never worked. That's why it's migrated here to use a token state change listener instead.

Reply:
Oh I see, I misunderstood why this broke. The initial state isn't overwritten. This wasn't working because without the setter, the token list isn't being initialized in the constructor, which is a bug. I'll fix that and restore these changes.

Reply:
Fixed here: https://github.com/MetaMask/core/pull/1505/commits/53340029ef82676cc6c4dc72fbfdc7329ffdd549